src/main/java/com/bio4j/model/uniprot/programs/ImportProteinInteractions.java

```diff
@@ -122,78 +123,106 @@ protected void importProteinInteractions(File inFile, File dbFolder) {
-        Element experiments = commentElem.getChild("experiments");
-        String intactId1St = interactant1.getAttributeValue("intactId");
-        String intactId2St = interactant2.getAttributeValue("intactId");
-        String organismsDifferSt = "";
-        String experimentsSt = "";
-        if (intactId1St == null) {
-          intactId1St = "";
-        }
-        if (intactId2St == null) {
-          intactId2St = "";
-        }
-        if (organismsDiffer != null) {
-          organismsDifferSt = organismsDiffer.getText();
-        }
-        if (experiments != null) {
-          experimentsSt = experiments.getText();
-        }
+    Note that because of 1. it would be easier if we passed the protein as an argument.
 
-        //----now we try to retrieve the interactant 2 accession--
-        String interactant2AccessionSt = interactant2.getChildText("id");
-        long protein2Id = -1;
-        if (interactant2AccessionSt != null) {
+    #### Remark
 
-          Optional<Protein<I,RV,RVT,RE,RET>> protein2Optional = graph.proteinAccessionIndex().getVertex(interactant2AccessionSt);
-
-          if(!protein2Optional.isPresent()) {
-
-            Optional<Isoform<I,RV,RVT,RE,RET>> isoformOptional = graph.isoformIdIndex().getVertex(interactant2AccessionSt);
-
-            if(isoformOptional.isPresent()) {
+    Again based on manual inspection it looks like protein-protein interactions are **not** *duplicated* in the UniProt XML file. I do not know if this implies a semantic direction or not. So, if you want to get all with which a given protein interacts, you need to get both inV and outV of protein-protein interaction edges.
+  */
+  private void importProteinInteractionsWithSource(
+    XMLElement entryXMLElem,
+    UniProtGraph<I,RV,RVT,RE,RET> graph
+  )
+  {
 
-              ProteinIsoformInteraction<I,RV,RVT,RE,RET> proteinIsoformInteraction = protein.addOutEdge(graph.ProteinIsoformInteraction(), isoformOptional.get());
-              proteinIsoformInteraction.set(graph.ProteinIsoformInteraction().experiments, experimentsSt);
-              proteinIsoformInteraction.set(graph.ProteinIsoformInteraction().organismsDiffer, organismsDifferSt);
-              proteinIsoformInteraction.set(graph.ProteinIsoformInteraction().intActId1, intactId1St);
-              proteinIsoformInteraction.set(graph.ProteinIsoformInteraction().intActId2, intactId2St);
+    /* We first get the protein from the entry xml element accession */
+    final Optional<Protein<I,RV,RVT,RE,RET>> optionalSrcProtein = graph.proteinAccessionIndex()
+      .getVertex(entryXMLElem.asJDomElement().getChildren(ENTRY.ACCESSION.element).get(0).getText());
+
+    optionalSrcProtein.ifPresent(
+      srcProtein -> {
+        entryXMLElem.asJDomElement().getChildren(ENTRY.COMMENT.element)
+          .stream()
+          .filter(
+            commentElem ->
+              commentElem.getAttributeValue(ENTRY.COMMENT.TYPE.attribute).equals(ENTRY.COMMENT.TYPE.INTERACTION)
```

Comment:
`ENTRY.COMMENT.TYPE.INTERACTION` here looks like a typo, because all the other things here have either `.element` or `.attribute` suffix. (I'm just saying that it _looks_ a bit cofusing)